packages/material-ui-lab/src/useAutocomplete/useAutocomplete.d.ts

```diff
@@ -210,7 +214,7 @@ export interface UseAutocompleteProps<
   /**
    * Array of options.
    */
-  options: T[];
+  options: Array<T | Option>;
```

Comment:
Do we even need to mention it? It seems that a sound type would be 
```suggestion
  options: Array<T | Option | string>;
```

But what about?

```suggestion
  options: T[];
```

Reply:
I was also wondering that but then if you are using typescript at least it will tell you the default supported type, no? I'm ok with having the type as it was since it is already marked as generic.

One thing is for sure - I don't really like the `options: Array<T | Option | string>;` option as to me it just adds another generic type in the list but technically it can also be a `number`.

I'm opened for suggestions - personally I would like if somewhere is indicated what the default type is.

Reply:
To be honest, I don't know what's best here. Maybe @eps1lon, our TypeScript owner, has a preference on this.

Reply:
Also, maybe we could have a look at the types of react-select and their history, they are in the same situation.

Reply:
Ok, maybe we can just have it as `AutocompleteOption[]` because the corresponding ticket is marked as Breaking Change ?

Reply:
react select uses a default value: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7caf9e949ed3779010d39927ee3f97f71708db7c/types/react-select/src/Select.d.ts#L56.

IMHO, it will be simpler to keep `options: T[];`, I don't see what we will gain by changing the type.

Reply:
After trying a couple of different ways I also think it would be better to keep it as it was but just add support for `{label: string}` data type out of the box.

Reply:
Sounds great, from my end.